synapse/replication/tcp/redis.py

```diff
@@ -297,12 +471,68 @@ def clientConnectionLost(self, connector: IConnector, reason: Failure) -> None:
         )
         super().clientConnectionLost(connector, reason)
 
-
 def format_address(address: IAddress) -> str:
     if isinstance(address, (IPv4Address, IPv6Address)):
         return "%s:%i" % (address.host, address.port)
     return str(address)
 
+def init_replication_factory(hs: HomeServer, connection: IRedisConnection, channel_names: List[str]) -> txredisapi.RedisFactory:
+    if isinstance(connection, SentinelRedisConnection):
+        factory = SynapseSentinelConnectionFactory(hs, connection._factory.sentinel_manager, connection._factory.service_name, True, connection)
+        factory.try_to_connect()
+    else:
+        factory = RedisDirectTcpReplicationClientFactory(
+            hs,
+            connection,
+            channel_names=self._channels_to_subscribe_to,
+        )
+        hs.get_reactor().connectTCP(
+            hs.config.redis.redis_host,
+            hs.config.redis.redis_port,
+            self._factory,
+            timeout=30,
+            bindAddress=None,
+        )
+    return factory
+
+class SynapseSentinelConnectionFactory(txredisapi.SentinelConnectionFactory):
+
+    maxDelay = 5
```

Comment:
Maybe you can store this in a constant or env. var or in homeserver.yaml?
Also, it might be beneficial to add a comment describing the unit of measure (assume it's seconds)